src/game/server/entities/character.cpp

```diff
@@ -611,7 +611,10 @@ void CCharacter::FireWeapon()
 
 		m_Core.m_Ninja.m_ActivationDir = Direction;
 		m_Core.m_Ninja.m_CurrentMoveTime = g_pData->m_Weapons.m_Ninja.m_Movetime * Server()->TickSpeed() / 1000;
-		m_Core.m_Ninja.m_OldVelAmount = length(m_Core.m_Vel);
+
+		// clamp to prevent massive MoveBox calculation lag with SG bug
+		float VelocityLength = length(m_Core.m_Vel);
+		m_Core.m_Ninja.m_OldVelAmount = std::clamp(VelocityLength, 0.0f, 6000.0f);
```

Comment:
```suggestion
		// clamp to prevent massive MoveBox calculation lag with shotgun bug
		m_Core.m_Ninja.m_OldVelAmount = std::clamp(length(m_Core.m_Vel), 0.0f, 6000.0f);
```
(suggestion, not needed for merge)